rando_modules/random_entrances.py

```diff
@@ -11,26 +11,48 @@
 from worldgraph import adjust
 
 from metadata.verbose_area_names import verbose_area_names
+from rando_enums.enum_ingame import StarSpirits
 
 def shuffle_dungeon_entrances(
     world_graph:dict,
     starway_spirits_needed_count:int,
+    required_star_spirits:list[int],
     shuffle_bowsers_castle:bool,
     write_spoilers:bool
 ) -> dict:
     if starway_spirits_needed_count > 6:
         shuffle_bowsers_castle = False
 
-    dungeon_border_node_ids = [
-        ("NOK_15/1", "TRD_00/0"), # Koopa Bros Fortress main
-        #("NOK_15/2", "TRD_00/4"), # Koopa Bros Fortress chest ledge
-        ("SBK_02/4", "ISK_01/0"), # Dry Dry Ruins
-        ("ARN_04/1", "DGB_00/0"), # Tubba Blubbas Castle
-        ("MAC_04/2", "OMO_03/4"), # Shy Guys Toybox
-        ("JAN_22/2", "KZN_01/0"), # Mt. Lavalava
-        ("MAC_01/5", "FLO_00/0"), # Flower Fields
-        ("SAM_10/1", "PRA_01/0"), # Crystal Palace
-    ]
+    dungeon_border_node_ids = []
+
+    if StarSpirits.ELDSTAR in required_star_spirits:
+        dungeon_border_node_ids.append(
+            ("NOK_15/1", "TRD_00/0") # Koopa Bros Fortress
+        )
+    if StarSpirits.MAMAR in required_star_spirits:
+        dungeon_border_node_ids.append(
+            ("SBK_02/4", "ISK_01/0") # Dry Dry Ruins
+        )
+    if StarSpirits.SKOLAR in required_star_spirits:
+        dungeon_border_node_ids.append(
+            ("ARN_04/1", "DGB_00/0") # Tubba Blubbas Castle
+        )
+    if StarSpirits.MUSKULAR in required_star_spirits:
+        dungeon_border_node_ids.append(
+            ("MAC_04/2", "OMO_03/4") # Shy Guys Toybox
+        )
+    if StarSpirits.MISSTAR in required_star_spirits:
+        dungeon_border_node_ids.append(
+            ("JAN_22/2", "KZN_01/0") # Mt. Lavalava
+        )
+    if StarSpirits.KLEVAR in required_star_spirits:
+        dungeon_border_node_ids.append(
+            ("MAC_01/5", "FLO_00/0") # Flower Fields
+        )
+    if StarSpirits.KALMAR in required_star_spirits:
+        dungeon_border_node_ids.append(
+            ("SAM_10/1", "PRA_01/0") # Crystal Palace
+        )
```

Comment:
This change to `shuffle_dungeon_entrances()` makes it so the dungeon shuffle gets restricted both during
* `RequireSpecificSpirits: True`, `LimitChapterLogic: False`, and
* `RequireSpecificSpirits: True`, `LimitChapterLogic: True`

If LCL is not turned on, but specific spirits are required, I think it's perfectly fine to have all dungeons in the shuffle, so please add a check against `logic_settings.limit_chapter_logic` for this somewhere sensible